### PR TITLE
zig: update to 0.9.0

### PR DESCRIPTION
**Pull request by linuxy**
Kqueue probably needs to be fixed as well.

### CODE REVIEW

File: os/posix.zig [hunk 1]
```diff
@@ -1,9 +1,10 @@
 const std = @import("std");
+const os = @import("os");
```
Comment (lithdew) [inline os/posix.zig]: Is `@import("os")` legal in the Zig 0.9.0?

Comment (linuxy) [reply os/posix.zig]: yeah  it works but the docs still recommend `const os = std.os;` I'll poke it. I'm including pike from a subdirectory in my current project. Just utilizing epoll+notifier at the moment, but looking good.

**Comment by lithdew** [comment]
Everything looks good so far; I'll try run it on a few machines tomorrow to double check everything's working properly and merge it right after. Thank you for the PR!

**Comment by lithdew** [comment]
Merged the PR in, and pushed an additional commit so that Pike's examples and syscalls works on Zig nightly as well. The examples are working on Linux and on a Mac laptop I borrowed; still need to see if I can get time to review the changes on Windows.

**Comment by linuxy** [comment]
> 
> 
> Merged the PR in, and pushed an additional commit commit so that Pike's examples and syscalls works on Zig nightly as well. The examples are working on Linux and on a Mac laptop I borrowed; still need to see if I can get time to review the changes on Windows.

Great. I just started playing with snow and saw that the socket api needed to be updated as well. You beat me to it!

**Pull request merged**